nextcord/client.py

```diff
@@ -2869,7 +3001,38 @@ class be provided, it should be a subclass of :py:class:`Interaction`.
         .. note::
 
             This is synchronous due to how slash commands are implemented.
-        """
+
+        Warns
+        --------
+        ~FutureWarning
+            Warns when the cls parameter is redefined to a custom class. This serves as a warning
+            in the event that defining this parameter was a mistake, and other parameters were to be defined instead.
+        """
+        # check if they are creating custom interaction subclasses
+        if cls is not Interaction:
```

Comment:
This warning seems a bit weird, the signature for `get_interaction` seems to imply you should pass a custom interaction when passing other custom interaction classes. The better check would be to look at the other classes and if none are custom, but that just seems a bit excessive IMO.
```python
class Client(nextcord.Client):
    def get_interaction(self, data: InteractionPayload):
        return super().get_interaction(
            data,
            application_autocomplete_interaction=CustomAutocompleteInteraction,
            application_interaction=CustomAppCmdInteraction,
            message_component_interaction=CustomComponentInteraction,
            modal_submit_interaction=CustomModalSubmitInteraction,
            cls=CustomInteraction,
        )
```